sdks/java/io/cassandra/src/test/java/org/apache/beam/sdk/io/cassandra/CassandraIOIT.java

```diff
@@ -0,0 +1,171 @@
+
+  @BeforeClass
+  public static void setup() throws Exception {
+    PipelineOptionsFactory.register(IOTestPipelineOptions.class);
+    options = TestPipeline.testingPipelineOptions()
+        .as(IOTestPipelineOptions.class);
+  }
+
+  @Test
+  public void testRead() throws Exception {
+    PCollection<Scientist> output = pipeline.apply(CassandraIO.<Scientist>read()
+        .withHosts(Collections.singletonList(options.getCassandraHost()))
+        .withPort(options.getCassandraPort())
+        .withKeyspace(CassandraTestDataSet.KEYSPACE)
+        .withTable(CassandraTestDataSet.READ_TABLE_NAME)
+        .withEntity(Scientist.class)
+        .withCoder(SerializableCoder.of(Scientist.class)));
+
+    PAssert.thatSingleton(output.apply("Count scientist", Count.<Scientist>globally()))
+        .isEqualTo(1000L);
+
+    PCollection<KV<String, Integer>> mapped =
+        output.apply(
+            MapElements.via(
+                new SimpleFunction<Scientist, KV<String, Integer>>() {
+                  public KV<String, Integer> apply(Scientist scientist) {
+                    KV<String, Integer> kv = KV.of(scientist.name, scientist.id);
+                    return kv;
+                  }
+                }
+            )
+        );
+    PAssert.that(mapped.apply("Count occurences per scientist", Count.<String, Integer>perKey()))
+        .satisfies(
+            new SerializableFunction<Iterable<KV<String, Long>>, Void>() {
+              @Override
+              public Void apply(Iterable<KV<String, Long>> input) {
+                for (KV<String, Long> element : input) {
+                  assertEquals(element.getKey(), 1000 / 10, element.getValue().longValue());
+                }
+                return null;
+              }
+            }
+        );
+
+    pipeline.run().waitUntilFinish();
+  }
+
+  @Test
+  public void testWrite() throws Exception {
+    String tableName = CassandraTestDataSet.createWriteDataTable(options);
+
+    ArrayList<Scientist> data = new ArrayList<>();
+    for (int i = 0; i < 1000; i++) {
+      Scientist scientist = new Scientist();
+      scientist.id = i;
+      scientist.name = "Name " + i;
+      data.add(scientist);
+    }
+
+    pipeline
+        .apply(Create.of(data))
+        .apply(CassandraIO.<Scientist>write()
+            .withHosts(Collections.singletonList(options.getCassandraHost()))
+            .withPort(options.getCassandraPort())
+            .withKeyspace(CassandraTestDataSet.KEYSPACE)
+            .withEntity(Scientist.class));
+
+    pipeline.run().waitUntilFinish();
+
+    Cluster cluster = CassandraTestDataSet.getCluster(options);
+    Session session = cluster.connect();
+    ResultSet result = session.execute("select id,name from " + CassandraTestDataSet.KEYSPACE
+        + "." + tableName);
+    List<Row> rows = result.all();
+    assertEquals(1000, rows.size());
+    for (Row row : rows) {
+      assertTrue(row.getString("name").matches("Name.*"));
+    }
```

Comment:
For the read test, it's fine to clean it up in test data set - we only create one table that's intended to stick around for a while.

For write tests however, as I read it, this code creates a new copy of the write table every time it runs - that means unless someone else comes around and manually cleans it up, this test leaks resources. Since the write table gets created during this test, it should also get cleaned up during this test. This is what the jdbc IO IT does - https://github.com/apache/beam/blob/master/sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/JdbcIOIT.java#L84t and what the elasticsearch IO IT does - https://github.com/apache/beam/blob/master/sdks/java/io/elasticsearch/src/test/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIOIT.java

Do you think that makes sense? I'd rather not require manual cleanup of resources that get created with every run of the tests since that can quickly add up.